diff --git a/common/config.py b/common/config.py
index 5d2271a1..39bc5745 100755
--- a/common/config.py
+++ b/common/config.py
@@ -19,7 +19,7 @@ from oslo_config import types
 from kolla.version import version_info as version
 
 
-BASE_OS_DISTRO = ['centos', 'rhel', 'ubuntu', 'oraclelinux', 'debian']
+BASE_OS_DISTRO = ['centos', 'rhel', 'ubuntu', 'oraclelinux', 'debian', 'openeuler']
 BASE_ARCH = ['x86_64', 'ppc64le', 'aarch64']
 DEFAULT_BASE_TAGS = {
     'centos': '7',
@@ -27,6 +27,7 @@ DEFAULT_BASE_TAGS = {
     'oraclelinux': '7-slim',
     'debian': 'stretch-backports',
     'ubuntu': '16.04',
+    'openeuler': '20.03-lts-sp2'
 }
 DISTRO_RELEASE = {
     'centos': '7',
@@ -34,6 +35,7 @@ DISTRO_RELEASE = {
     'oraclelinux': '7',
     'debian': 'stretch-backports',
     'ubuntu': '16.04',
+    'openeuler': '20'
 }
 
 # This is noarch repository so we will use it on all architectures
@@ -222,7 +224,7 @@ _CLI_OPTS = [
                      ' push properly')),
     cfg.IntOpt('retries', short='r', default=3, min=0,
                help='The number of times to retry while building'),
-    cfg.MultiOpt('regex', types.String(), positional=True,
+    cfg.MultiOpt('regex', types.String(), positional=True, required=False,
                  help=('Build only images matching regex and its'
                        ' dependencies')),
     cfg.StrOpt('registry',
diff --git a/image/build.py b/image/build.py
index e5eb8dd0..1f5a20a8 100755
--- a/image/build.py
+++ b/image/build.py
@@ -195,6 +195,29 @@ SKIPPED_IMAGES = {
         # TODO(jeffrey4l): remove tripleo-ui when following bug is fixed
         # https://bugs.launchpad.net/tripleo/+bug/1744215
         "tripleo-ui"
+    ],
+    'openeuler+binary': [
+        "almanach-base",
+        "bifrost-base",
+        "blazar-base",
+        "dragonflow-base",
+        "freezer-base",
+        "karbor-base",
+        "kuryr-base",
+        "monasca-base",
+        "neutron-bgp-dragent",
+        "nova-mksproxy",
+        "ovsdpdk",
+        "searchlight-base",
+        "senlin-base",
+        "solum-base",
+        "tripleo-ui",
+        "vmtp",
+        "zun-base",
+    ],
+    'openeuler+source': [
+        "ovsdpdk",
+        "tripleo-ui"
     ]
 }
 
@@ -678,7 +701,7 @@ class KollaWorker(object):
                              conf.rpm_setup_config if repo_file is not None])
         self.rpm_setup = self.build_rpm_setup(rpm_setup_config)
 
-        rh_base = ['centos', 'oraclelinux', 'rhel']
+        rh_base = ['centos', 'oraclelinux', 'rhel', 'openeuler']
         rh_type = ['source', 'binary', 'rdo', 'rhos']
         deb_base = ['ubuntu', 'debian']
         deb_type = ['source', 'binary']
